salt/states/file.py

```diff
@@ -677,6 +677,16 @@ def _delete_not_kept(nfn):
                 os.path.relpath(nfn, root), None, exclude_pat
             ):
                 return
+            # Before we can accurately assess the removal of a file, we must
+            # check for windows case sensitive files. If we originally meant
+            # to keep a file, but due to case sensitivity python would otherwise
+            # remove the file, check against the original list.
+            if win_keep:
+                for item in win_keep:
+                    if item.lower() != nfn.lower():
+                        continue
+                    elif item.lower() == nfn.lower():
+                        return
```

Comment:
**question** is the continue part necessary? Or could this even just be written like so:

```
for item in win_keep or []:
    if item.lower() == nfn.lower():
        return
```

Reply:
Also, looking  at the other bit, maybe just:

```
for item in copy.deepcopy(keep):  # not sure if the copy is even necessary?
    if item.lower() == nfn.lower():
        return
```

Reply:
(then win_keep the parameter wouldn't be necessary)

Reply:
Yeah looking at this again, I believe the param is needed. The function that does the "deciding" of the files to keep or not to keep on the system need a case-sensitive version. but i can take out the continue